### PR TITLE
feat: [TEDIX-627] Parent e-mail is optional in the Registration form

### PR DESCRIPTION
**Pull request by DmytroNefyodov**
**Description:**Parent e-mail is optional in the Registration form

**Youtrack:** https://youtrack.raccoongang.com/agiles/104-268/105-2572?issue=T-627

### CODE REVIEW

File: common/djangoapps/tedix_ro/forms.py [hunk 1]
```diff
@@ -162,7 +163,7 @@ def clean_parent_phone(self):
         user = User.objects.filter(email=parent_email).first() if parent_email else None
         # TODO: there is a validator field, remove the additional validation of
         # the length of the phone number in the future
-        if len(parent_phone) < 10 or len(parent_phone) > 15:
+        if len(parent_phone) != 0 and len(parent_phone) < 10 or len(parent_phone) > 15:
```
Comment (sendr) [inline common/djangoapps/tedix_ro/forms.py:166]: is this change really needed? It seems to me that it's the same case as it was before...

Comment (DmytroNefyodov) [reply common/djangoapps/tedix_ro/forms.py:166]: for an optional field, this condition is needed, if we leave an empty field and click on Create Account, we will get a ValidationError(_('The parent phone number length must be from 10 to 15 digits.')

Comment (sendr) [reply common/djangoapps/tedix_ro/forms.py:166]: ok, then can it be like this `parent_phone and len(parent_phone) < 10 or len(parent_phone) > 15`?

File: common/djangoapps/tedix_ro/tasks.py [hunk 2]
```diff
@@ -190,8 +190,13 @@ def send_student_extended_reports(course_report_id):
             'email_from_address',
             settings.DEFAULT_FROM_EMAIL
         )
+        mail_recipients = [user.email,]
+
         parent = user.studentprofile.parents.first()
-        send_mail(subject, txt_message, from_address, [user.email, parent.user.email], html_message=html_message)
+        if parent and parent.user.email:
```
Comment (sendr) [inline common/djangoapps/tedix_ro/tasks.py:196]: is there a case when parent's user doesn't have an email?

File: common/djangoapps/tedix_ro/tasks.py [hunk 3]
```diff
@@ -208,9 +213,13 @@ def send_student_extended_reports(course_report_id):
             'sms/light_report.txt',
             context
         )
+        parent_phone = student_course_report.student.parent_phone
```
Comment (sendr) [inline common/djangoapps/tedix_ro/tasks.py:216]:
```suggestion
        parent_phone = user.studentprofile.parent_phone
```

**Review by sendr — APPROVED** [review]
